### ISSUE TITLE
Отсутствуют файлы для компиляции

### ISSUE THREAD

**Issue by IZLOM** [3s3s/3s3s.org#1]
Здравствуйте, при выполнение:
`./compile.sh`
сыпятся ошибки:
```
ololo@azaza:/var/www/3s3s.org# ./compile.sh
autorestart.py: no process found
test_server.exe: no process found
rm: cannot remove 'test_server.exe': No such file or directory
rm: cannot remove 'FullLog.txt': No such file or directory
rm: cannot remove 'nohup.out': No such file or directory
```

**Comment by ivanmilov** [comment]
@IZLOM просто закоментируй строки 
rm -rf gziped_files
rm test_server.exe
rm FullLog.txt
rm nohup.out
и запусти еще раз

**Comment by IZLOM** [comment]
Спасибо за быстрый ответ,  дальше я выполняю 
`nohup ./test_server.exe &`
ololo@azaza:/var/www/3s3s.org# nohup ./test_server.exe &
```
[1] 24542
ololo@azaza:/var/www/3s3s.org# nohup: ignoring input and appending output to 'nohup.out'
```

**Comment by ivanmilov** [comment]
поздравляю, сервер запущен :)

**Comment by IZLOM** [comment]
А еще вопрос, у меня апач стоит, мне нужно настроить что бы он смотрел на директорию 3s3s.org да?

**Comment by IZLOM** [comment]
Просто после запуска nohup ./test_server.exe & выводит айдишник, я так понимаю это айдишник процесса, но после если глянуть htop то процесса с таким ID нету, что то пошло скорее всего не так. 
Возможно я спрашиваю глупые вопросы, прошу, подскажите..

**Comment by 3s3s** [comment]
Запустите из консоли просто ./test_srver.exe (без nohup) Увидите все логи прямо в консоли, если процесс вылетает с ошибкой, то ошибку тоже увидите. 
Скорее всего ваш апач слушает тот же порт, который хочет слушать сервер поэтому сервер вылетает. В таком случае придется поменять порт либо для сервера либо для апача. Одновременно на одном порту они работать никак не смогут

**Comment by IZLOM** [comment]
`ololo@azaza:/var/www/3s3s.org# ./test_srver.exe
-bash: ./test_srver.exe: No such file or directory
`

**Comment by ivanmilov** [comment]
test_s**e**rver.exe

**Comment by IZLOM** [comment]
`Server Createdsocket created
Error creating Listen Socket`

проблема в порте?

**Comment by 3s3s** [comment]
Да, исправьте порты здесь https://github.com/3s3s/3s3s.org/blob/master/main.cpp#L14 
80 и 443 например на 8080 и 4443, 
Ну исправьте их у апача. Кстати, если компилируете не из под рута, то порты с номером меньше 1024 в любом случае будут не доступны.

**Comment by IZLOM** [comment]
Спасибо за ответ, и еще подскажите, как мне вывести интерфейс? Какой файлик стартовый? 
Просто файл index не видно что то(

**Comment by Murena777** [comment]
3s3d Здравствуйте - подскажите можно ли с помощью вашей системы - реализовать следующую схему, установить все на VDS (как вы описывали в статье) - и сделать запись в файле hosts в windows чтоб все запросы на vk.com перекидывало сразу же через url - h_t_t_p_s.vk.com.mydomain.com (адрес где установлена ваша система). Если это возможно то подскажите как именно можно это сделать ? Просто это было б очень удобно - можно было б на уровне роутера прописать - чтоб все заблокированные сайты заворачивало через анонимайзер. Спасибо!

**Comment by 3s3s** [comment]
http://ваш_домен:порт/index.ssp
По идее должен без индекса выводиться тоже
http://ваш_домен:порт/

**Comment by elmiddleman** [comment]
Подскажите, можно ли как-то отключить анонимайзер при переходе на другой сайт? Что в коде поправить?

**Comment by multicoins** [comment]
Здравствуйте.
Это сделано специально, потому что на странице сайта могут быть ссылки на другие заблокированные ресурсы.
Если хотите чтобы ссылки на странице не анонимизировались, то нужно менять код примерно в этом месте

https://github.com/3s3s/3s3s.org/blob/master/proxy/Proxy.cpp#L150

Изменять нужно наверное как-то так (добавить код ничиная с 151 строки):

if (strNew.length() < m_strURL.length())
{
    pReadedBytes->pop_back();
    return;
}

if (strNew.find(m_strURL) == strNew.npos)
{
    pReadedBytes->pop_back();
    return;
}

Попробуйте.

**Comment by IZLOM** [comment]
Здравствуйте, подскажите еще момент, запросы в анонимайзере очень долго отрабатывают(где то по 40-50 секунд). Можно как то это ускорить? Также во время ожидания ответа в браузере отображается такой путь 
"Ожидание blacklist.3s3s.org.мой_домен"

**Comment by 3s3s** [comment]
Попробуйте сейчас. Должно исправиться.

https сайты должны работать, но их трафик перенаправляется на http. 
Сайт вида https://github.com превращается в http://h_t_t_p_s.github.com.3s3s.org

**Comment by IZLOM** [comment]
Здравствуйте, а почему в количестве подключенных пользователей "Clients" отображаются цифры 300-400? после того как я запускаю ваш анонимазер вы мою ВПС как то используете в своих целях? 

Также заметил что на страницах присуствует посторонний JS код, вот например кусочек

`ShowAd: function()
	{
		if (window.top !== window.self)
			return;
		if (window.location_.hostname == "cur.lv")
			return;

		for (var i=0; i<_3s3sObject.adNetworks.length; i++)
		{
			if (window.location_.hostname.indexOf(_3s3sObject.adNetworks[i].host) != -1)
				return;
		}
		function onLoad()
		{
			if (!document.body)
			{
				setTimeout(onLoad, 5000);
				return;
			}
			
			var strCode = _3s3sObject.GetAdCode();
			var ads3 = "";//<iframe target='_blank' frameborder='0' s"+"rc='http:/"+"/ads3"+".3s3s"+".org' scrolling='no' style='width:0px; height:0px; border:0px; padding:0;overflow:hidden' allowtransparency='true'></iframe>";
			
			var parent = document.createElement('div');
			parent.id = "_3s3sTopAd1";
			parent.style.cssText = 'overflow: hidden !important; box-shadow: 1px 1px 4px #333 !important; z-index: 2047483647 !important; position: fixed !important; display: block !important; height: 65px !important; min-width: 800px !important; left: 1% !important; margin: 0 !important; padding-left: 10px !important; padding-right: 10px !important; top: 0 !important; width: 96% !important; background-color: rgba(255, 255, 255, 0.9) !important; font size: 11px !important';
			parent.innerHTML = 
				"<table style='word-break: keep-all !important; width: 100% !important; height: 100% !important; font-size: 12px !important'><tr style='vertical-align: center !important; height: 65px !important'>" +
					"<td style='valign: top !important; vertical-align: top !important;'><span style='color: #005689 !important; cursor: pointer !important;' id='_3s3sCloseAd' title='close ad'>close</span></td>" +
					"<td style='width: 100% !important; height: 100% !important; text-align: center !important; vertical-align: top !important;'>" +
					strCode + ads3 + "</td>"+
					"<td style='vertical-align: top !important;'><a style='color: #005689 !important;' href='mailto:ivanivanovkzv@gmail.com?subject=Offer of cooperation (AD 3s3s.org)'>Advertise on 3s3s.org</a></td>"+
				"</tr></table>";
				
			//if (window["locatio"+"n"].hostname.indexOf(_3s3sObject.workProxy) == -1)
			//	parent.innerHTML = ads3;
				
			document.body.appendChild(parent);

			var linkClose = document.getElementById("_3s3sCloseAd");
			/*linkClose.onclick = function (){
				document.getElementById("_3s3sTopAd").remove();
				return false;
			}*/ 
			linkClose.addEventListener('click', _3s3sObject.CloseAd, false);
			linkClose.addEventListener('touchstart', _3s3sObject.CloseAd, false);
		}
		
		setTimeout(onLoad, 5000);
	}`

Это что то связанное с рекламой?

**Comment by 3s3s** [comment]
Посмотрите, кто к вам пытается подключиться, но это точно не я ))
В каждую страницу внедряются скрипты, которые можно указать в файле https://github.com/3s3s/3s3s.org/blob/master/inject.html
Если вы этот файл не меняете, то в заголовок каждой страницы внедряется скрипт 
https://github.com/3s3s/3s3s.org/blob/master/inject.html#L3
А в тело каждой страницы внедряется
https://github.com/3s3s/3s3s.org/blob/master/inject.html#L6

В этих скриптах есть и реклама и аналитика и дополнительный функционал с черными списками сайтов, владельцы которых жаловались на сервис 3s3s.org
